datasafe-document-api/src/main/java/de/adorsys/datasafe/business/api/deployment/document/DocumentWriteService.java

```diff
@@ -1,12 +1,17 @@
 package de.adorsys.datasafe.business.api.deployment.document;
 
-import de.adorsys.datasafe.business.api.types.action.WriteRequest;
-
 import java.io.OutputStream;
 
+import de.adorsys.datasafe.business.api.types.action.ObjectMetadata;
+import de.adorsys.datasafe.business.api.types.action.WriteRequest;
+
 public interface DocumentWriteService {
 
     OutputStream write(WriteRequest request);
+
+    default OutputStream write(WriteRequest request, ObjectMetadata metadata) {
```

Comment:
@francis-pouatcha 
I'm doing some similar work in
https://github.com/adorsys/datasafe/tree/feature/DOC-122_Lean-filesytem-interface
and there one can write metainformation into storage controlled by himself using WriteRequest that has interface based ResourceLocation, so that he can implement interface and extend it with metainfo and store it using his own storage system by i.e.
```
if (location instanceof ResourceLocationWithMeta) {
  Meta meta = location.getMeta();
  storage.store(data, meta);
}
```

Reply:
See 
https://github.com/adorsys/datasafe/pull/48
there PrivateResourceWithMeta - it is used for writing 'metainformation'